src/plugins/index_pattern_editor/public/components/index_pattern_editor_flyout_content.tsx

```diff
@@ -223,13 +230,14 @@ const IndexPatternEditorFlyoutContentComponent = ({
       rollupIndex,
       type,
       matchedIndices.exactMatchedIndices,
+      isLoadingMatchedIndices,
     ]
   );
 
   useEffect(() => {
     loadTimestampFieldOptions(title);
     getFields().timestampField?.setValue('');
-  }, [matchedIndices, loadTimestampFieldOptions, title, getFields]);
+  }, [loadTimestampFieldOptions, title, getFields]);
```

Comment:
Nice. I am surprised you don't get the `eslint` error when you add unnecessary dependency. Did you disable the rule for this plugin?

Reply:
Nope. don't see any overrides. 

I saw the rule complaint where there is a dependency missing, but maybe it doesn't complain when there is an unnecessary dependency?  (I can imagine there could be use-cases where it is usefull)

Reply:
Interesting, I just realised that it does not pass `eslint react-hooks/exhaustive-deps` rule if you add an unnecessary dependency to a `useCallback` but not to `useEffect`. I guess it makes (_kinda_) sense 😄